### PR TITLE
Introduce specific dispatcher

### PR DESCRIPTION
**Pull request by arothuis**
[133173121](https://www.pivotaltracker.com/story/show/133173121)

This PR merely introduces the concept of a specific event dispatcher that can be called from the Replay CLI command, using a given EventCollection and ProjectorCollection. I've outlined the planned approach in the comments in the try/catch block.

Branched from #188

### CODE REVIEW

**Comment by arothuis** [comment]
Rebased on develop.

File: src/Surfnet/StepupMiddleware/MiddlewareBundle/Service/SpecificEventDispatcher.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+namespace Surfnet\StepupMiddleware\MiddlewareBundle\Service;
+
+use Exception;
+use Surfnet\StepupMiddleware\MiddlewareBundle\EventSourcing\DBALEventHydrator;
+use Surfnet\StepupMiddleware\MiddlewareBundle\EventSourcing\EventCollection;
+use Surfnet\StepupMiddleware\MiddlewareBundle\EventSourcing\ProjectorCollection;
+
+final class SpecificEventDispatcher
```
Comment (DRvanR) [inline src/Surfnet/StepupMiddleware/MiddlewareBundle/Service/SpecificEventDispatcher.php]: "a specific"  - as in, say, EventReplayOnly :wink:

File: src/Surfnet/StepupMiddleware/MiddlewareBundle/Service/SpecificEventDispatcher.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * @var DBALEventHydrator
+     */
+    private $eventHydrator;
+
+    public function __construct(DBALConnectionHelper $connectionHelper, DBALEventHydrator $eventHydrator)
+    {
+        $this->connectionHelper = $connectionHelper;
+        $this->eventHydrator = $eventHydrator;
```
Comment (DRvanR) [inline src/Surfnet/StepupMiddleware/MiddlewareBundle/Service/SpecificEventDispatcher.php]: Why does it need this?

File: src/Surfnet/StepupMiddleware/MiddlewareBundle/Service/SpecificEventDispatcher.php (hunk 1 above)
Comment (arothuis) [inline src/Surfnet/StepupMiddleware/MiddlewareBundle/Service/SpecificEventDispatcher.php]: I would like to call it an `EventStreamReplayer` but we already have one.

File: src/Surfnet/StepupMiddleware/MiddlewareBundle/Service/SpecificEventDispatcher.php (hunk 2 above)
Comment (arothuis) [inline src/Surfnet/StepupMiddleware/MiddlewareBundle/Service/SpecificEventDispatcher.php]: In order to get the selected events from the event stream as objects.

**Comment by arothuis** [comment]
I could rename the EventCollection to an EventNameCollection, create an EventCollection that will contain the hydrated events. Call the DBALEventHydrator outside the EventDispatcher and dispatch using the hydrated events.

Then again, if we would rename the Dispatcher to an EventReplayer, I'd say it behaves as expected (just like the original EventReplayer).

**Comment by arothuis** [comment]
Changes are added to later branch. Closing and deleting this PR.

**Pull request closed without merging**